### PR TITLE
vala-language-server: init at 0.48.1

### LINKED ISSUE #85024: Package request: vala-language-server
**Project description**
Code Intelligence for Vala 

**Metadata**

* homepage URL: https://github.com/benwaffle/vala-language-server
* source URL: https://github.com/benwaffle/vala-language-server
* license: LGPL2.1
* platforms: unix, linux, darwin, ...

### PR DESCRIPTION
**Pull request by felix-andreas**
Fixes #85024

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions (elementary OS)
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by andreasfelix** [comment]
Hi, I recently started using Nix and adding the [vala-language-server](https://github.com/benwaffle/vala-language-server) seemed like a useful first exercise. The `default.nix` file is loosely based on https://github.com/NixOS/nixpkgs/pull/83386. I did my best to match the [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md). For now I added myself as only maintainer, but as I am a new user, I am unsure if I can guarantee to maintain it. So maybe someone else wants to add themselves to the maintainers list?

File: pkgs/development/tools/vala-language-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{ stdenv
+, fetchFromGitHub
+, meson
+, ninja
+, pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/development/tools/vala-language-server/default.nix:5]:
```suggestion
, pkg-config
```

File: pkgs/development/tools/vala-language-server/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "vala-language-server";
+  version = "0.48.1";
+  repoName = pname;
+
+  src = fetchFromGitHub {
+    owner = "benwaffle";
+    repo = repoName;
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/vala-language-server/default.nix:22]:
```suggestion
  src = fetchFromGitHub {
    owner = "benwaffle";
    repo = pname;
    rev = version;
```

File: pkgs/development/tools/vala-language-server/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    json-glib
+    jsonrpc-glib
+    vala
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Code Intelligence for Vala & Genie";
+    homepage = "https://github.com/benwaffle/vala-language-server";
+    license = licenses.lgpl21;
+    maintainers = [ "Felix Andreas <fandreas@physik.hu-berlin.de>" ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/vala-language-server/default.nix:45]:
```suggestion
    maintainers = with maintainers; [ andreasfelix ];
```

and add yourself to tha maintainers file.

Comment (andreasfelix) [reply pkgs/development/tools/vala-language-server/default.nix]: thank you! i have implemented you suggestions.

**Comment by worldofpeace** [comment]
Oh hi @andreasfelix, I recall some of contributions in the elementary community. Cool to see u here, I maintain pantheon and vala here.

**Comment by worldofpeace** [comment]
> So maybe someone else wants to add themselves to the maintainers list?

Feel free to add me, I wasn't really sure if gvls or vala-language-server was going to be defacto or something (like will Code use this one day).
You can also add the following to the expression to get semi-automatic updates
```
passthru = {
  updateScript = nix-update-script {
    attrPath = pname;
  };
}
```

and you can run
```
nix-shell maintainers/scripts/update.nix --argstr maintainer andreasfelix  --argstr commit true
```

and it will make a commit for you if there's a change upstream. See also https://nixos.org/manual/nixpkgs/stable/#ssec-stdenv-attributes `passthru.updateScript`

File: pkgs/development/tools/vala-language-server/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+, libgee
+, json-glib
+, jsonrpc-glib
+, vala
+}:
+
+stdenv.mkDerivation rec {
+  pname = "vala-language-server";
+  version = "0.48.1";
+  src = fetchFromGitHub {
```
Comment (worldofpeace) [inline pkgs/development/tools/vala-language-server/default.nix:17]: can you add whitespace here 
```suggestion

  src = fetchFromGitHub {
```

File: pkgs/development/tools/vala-language-server/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    libgee
+    json-glib
+    jsonrpc-glib
+    vala
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Code Intelligence for Vala & Genie";
+    homepage = "https://github.com/benwaffle/vala-language-server";
+    license = licenses.lgpl21;
```
Comment (worldofpeace) [inline pkgs/development/tools/vala-language-server/default.nix:42]: this should be `lgpl21Plus` https://github.com/benwaffle/vala-language-server/blob/master/meson.build#L3

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Hi, overall copying the other expression and the dependency declaration is all good.

Could we make additional changes as improvements?
* gnome-builder support would be cool https://github.com/benwaffle/vala-language-server/blob/master/plugins/gnome-builder/meson.build

It adds the following native dependencies https://github.com/benwaffle/vala-language-server/blob/master/plugins/gnome-builder/meson.build#L7

**Comment by worldofpeace** [comment]
> Result of `nixpkgs-review pr 102664` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 1 package failed to build:
> 
> ```
> Found ninja-1.10.1 at /nix/store/ba77flp1iya22w2v7yih1vwz8z4zwdnb-ninja-1.10.1/bin/ninja
> Using 'STRIP' from environment with value: 'strip'
> Using 'STRIP' from environment with value: 'strip'
> meson: enabled parallel building
> @nix { "action": "setPhase", "phase": "buildPhase" }
> building
> build flags: -j8 -l8
> ^M[0/34] Generating vala-language-server.1 with a custom commandninja: fatal: posix_spawn: No such file or directory
> ```

Interesting, it passed on ofborg. Will try to reproduce

**Comment by worldofpeace** [comment]
> gnome-builder support would be cool 

I looked at the installed output and there's a plugin. But it assumes ABI is 3.38, so we'd need to add those dependencies.

As for the nixpkgs-review result, I can't reproduce

File: pkgs/development/tools/vala-language-server/default.nix (hunk 5 above)
Comment (andreasfelix) [reply pkgs/development/tools/vala-language-server/default.nix]: thanks! fixed it

**Comment by andreasfelix** [comment]
> Oh hi @andreasfelix, I recall some of your contributions in the elementary community. Cool to see u here, I maintain pantheon and vala here.

Yeees, hi there 👋 Thanks for the fast and detailed reply!

I incorporate your suggestion with the update script and also added you as additional maintainer.

>  gnome-builder support would be cool 

done

> It adds the following native dependencies https://github.com/benwaffle/vala-language-server/blob/master/plugins/gnome-builder/meson.build#L7

Is it save to assume that `sh` is available? I added the dash shell dependency for now.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102664` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vala-language-server</li>
  </ul>
</details>

File: pkgs/development/tools/vala-language-server/default.nix [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    };
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    scdoc
+    # GNOME Builder Plugin
+    dash
```
Comment (worldofpeace) [inline pkgs/development/tools/vala-language-server/default.nix:41]: the nix sandbox has /bin/sh, so I wonder if find_program https://github.com/benwaffle/vala-language-server/blob/master/plugins/gnome-builder/meson.build#L7 considers /bin, or maybe we'd have to do
```
preBuild = ''
  # sh for gnome-builder-plugin
  export PATH=$PATH:/bin
'';

Comment (andreasfelix) [reply pkgs/development/tools/vala-language-server/default.nix]: Seems to work without it: `Program sh found: YES`

Comment (andreasfelix) [reply pkgs/development/tools/vala-language-server/default.nix]: I am not sure if I used sand-boxing tough

**Review by worldofpeace — APPROVED** [review]
This LGTM. I checked the ofborg build logs and it said it could find sh with sandboxing.
Could you please fixup and rebase your history so that we have two commits?
```
maintainers: add andreasfelix
vala-language-server: init at 0.48.1
```

**Comment by andreasfelix** [comment]
done

**Comment by worldofpeace** [comment]
Thx for contributing @andreasfelix :sparkles:

**Pull request merged**